akkeeper/src/main/scala/akkeeper/storage/zookeeper/async/ZookeeperInstanceStorage.scala

```diff
@@ -50,23 +54,40 @@ private[akkeeper] class ZookeeperInstanceStorage(config: ZookeeperClientConfig)
       .map(fromBytes[InstanceInfo])
   }
 
+  //akka-http default timeout is 20 seconds, so ensure that the call does not exceed it
+  //timeout outer future in 19 seconds
   override def getInstances: Future[Seq[InstanceId]] = {
-    val instancesFuture = for {
+    val t0 = System.nanoTime()
+    val instancesFuture: Future[Seq[Future[Seq[String]]]] = for {
+      //timeout
       containers <- zookeeperClient.children("")
     } yield for {
       container <- containers
     } yield zookeeperClient.children(container)
-    instancesFuture
+    val expectedExecution = instancesFuture
       .flatMap(f => Future.sequence(f).map(_.flatten))
       .map(_.map(pathToInstanceId))
       .recover(notFoundToEmptySeq[InstanceId])
+    expectedExecution.onComplete { result:Try[Seq[InstanceId]] =>
```

Comment:
Is this future going to be completed on akka-http timeout?

Reply:
Yes - it is independent. Akka-http waits for completion of http request for `akka.http.server.request-timeout`
This future will be completed at unknown point. if it is completed within that timeout, result will be set back to http layer and completes the request. Otherwise, request is completed with timeout, but the Future computation here continues and eventually it should finish, logging the result.